webrtc/stun/stun_utils.nim

```diff
@@ -0,0 +1,57 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import sequtils, typetraits, std/sha1
+import bearssl
+
+proc generateRandomSeq*(rng: ref HmacDrbgContext, size: int): seq[byte] =
+  result = newSeq[byte](size)
+  rng[].generate(result)
+
+proc createCrc32Table(): array[0..255, uint32] =
+  for i in 0..255:
+    var rem = i.uint32
+    for j in 0..7:
+      if (rem and 1) > 0:
+        rem = (rem shr 1) xor 0xedb88320'u32
+      else:
+        rem = rem shr 1
+    result[i] = rem
```

Comment:
I'd recommend to avoid `result` and use explicit returns unless you have a very strong preference for that.

Reply:
Hum.... I like result actually, it's a neat tool

Reply:
I find it unusual and harder to reason. Another reason is that it is used arbitrarily in the procs. I believe it is better to follow the same pattern.